packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -416,36 +416,129 @@ class DaemonDomain extends Domain {
   /// is correct.
   Future<Map<String, Object>> getSupportedPlatforms(Map<String, Object?> args) async {
     final String? projectRoot = _getStringArg(args, 'projectRoot', required: true);
-    final List<String> result = <String>[];
+    final List<String> platforms = <String>[];
+    final Map<String, Object> platformTypesMap = <String, Object>{};
     try {
       final FlutterProject flutterProject = FlutterProject.fromDirectory(globals.fs.directory(projectRoot));
       final Set<SupportedPlatform> supportedPlatforms = flutterProject.getSupportedPlatforms().toSet();
-      if (featureFlags.isLinuxEnabled && supportedPlatforms.contains(SupportedPlatform.linux)) {
-        result.add('linux');
-      }
-      if (featureFlags.isMacOSEnabled && supportedPlatforms.contains(SupportedPlatform.macos)) {
-        result.add('macos');
-      }
-      if (featureFlags.isWindowsEnabled && supportedPlatforms.contains(SupportedPlatform.windows)) {
-        result.add('windows');
-      }
-      if (featureFlags.isIOSEnabled && supportedPlatforms.contains(SupportedPlatform.ios)) {
-        result.add('ios');
-      }
-      if (featureFlags.isAndroidEnabled && supportedPlatforms.contains(SupportedPlatform.android)) {
-        result.add('android');
-      }
-      if (featureFlags.isWebEnabled && supportedPlatforms.contains(SupportedPlatform.web)) {
-        result.add('web');
-      }
-      if (featureFlags.isFuchsiaEnabled && supportedPlatforms.contains(SupportedPlatform.fuchsia)) {
-        result.add('fuchsia');
-      }
-      if (featureFlags.areCustomDevicesEnabled) {
-        result.add('custom');
+
+      void handlePlatformType(
+        PlatformType platform,
+      ) {
+        bool isSupported = true;
+        bool canBeEnabledWithCreate = true;
+        final List<String> reasons = <String>[];
+        switch (platform) {
+          case PlatformType.linux:
+            if (!featureFlags.isLinuxEnabled) {
+              isSupported = false;
+              reasons.add('the Linux feature is not enabled ("flutter config --enable-linux-desktop")');
+              canBeEnabledWithCreate = false;
```

Comment:
I had a quick look in Dart-Code whether we handle this, and it turns out we do:

https://github.com/Dart-Code/Dart-Code/blob/cd0806c1a2f34136c111cbc543217aa9569190c5/src/extension/flutter/flutter_daemon.ts#L224-236

So I wonder if we should:

1. Change this to `canBeEnabled` and let the client handle it
2. Add a second flag for `canBeEnabledWithConfig`
3. Merge the flags into `reasons`, something like:
```js
  "reasons": [
    {
        "reason": "The Linux feature is not enabled (...)",
        "code": "flutterCreate" // Some well known reason codes that the editor could understand ("flutterCreate", "flutterConfig") or similar
    }
  ]
```

If we did (3), we could also split the reason and the fix so that the UI could be rendered better (for example we can not show the command text if we will do this for the user):

```js
    {
        "reasonText": "The Linux feature is not enabled",
        "fixText": "Run 'flutter config --enable-linux-desktop'",
        "fixCode": "flutterCreate" // Some well known reason codes that the editor could understand ("flutterCreate", "flutterConfig") or similar
    }
```

Reply:
Let's go with option 3

Reply:
done